pages/anime/[id].tsx

```diff
@@ -0,0 +1,117 @@
+import {useRouter} from "next/router";
+import Head from "next/head";
+import Image from "next/image";
+import {GetServerSideProps, NextPage} from "next";
+import GlobalStyle, {Container} from "../../styles/global.styled";
+import BackToHomePageIcon from '../../assets/icons/backArrow.svg';
+import StatusIcon from '../../assets/icons/status.svg';
+import {
+  AnimeImage,
+  AnimeImageContainer,
+  AnimeInfoBlock,
+  AnimeInfoDescription,
+  AnimeInfoMainContent,
+  AnimeInfoTitle,
+  AnimeInfoWrapper,
+  BackMainPage,
+  BackMainPageText,
+  CharacteristicsAnime,
+  Content,
+  InfoDescription,
+  InfoDescriptionTitle,
+  Page,
+  Rating,
+  RatingBlock,
+  RatingItem,
+  RatingTitle,
+  StatusIconStyles,
+  Title
+} from "../../styles/animePage.styled";
+import {AnimePageProps} from "../../interfaces/animePage";
+import {AnimeTypeList} from "../../components/animeTypeList";
+
+const AnimePage: NextPage<AnimePageProps> = ({data}) => {
+  const route = useRouter()
+
+  const {mal_id, popularity, images, status, title, type, source, episodes, score, rank} = data
+
+  return (
+    <div>
+      <Head>
+        <title>Anime Page</title>
+        <meta name="Anime Page" content="Anime"/>
+        <link rel="icon" href="/favicon.ico"/>
+      </Head>
+      <GlobalStyle/>
+      <Page>
+        <Container>
+          <Content>
+            <BackMainPage onClick={() => route.push('/')}>
+              <Image src={BackToHomePageIcon} alt='arrow-icon'/>
+              <BackMainPageText>Go back to Main</BackMainPageText>
+            </BackMainPage>
+            <AnimeInfoWrapper key={mal_id}>
+              <AnimeInfoMainContent>
+                <AnimeImageContainer>
+                  <AnimeImage src={images?.jpg.large_image_url} alt='image-anime'/>
+                </AnimeImageContainer>
+                <AnimeInfoBlock>
+                  <AnimeInfoTitle>
+                    <Title>{title} <StatusIconStyles src={StatusIcon.src} alt='status-icon'/> </Title>
+                    <CharacteristicsAnime>
+                      <AnimeTypeList type={type} episodes={episodes} source={source} status={status}/>
+                    </CharacteristicsAnime>
+                    <RatingBlock>
```

Comment:
this
                    <RatingBlock>
                      <RatingItem>
                        <Rating>{score}</Rating>
                        <RatingTitle>Score</RatingTitle>
                      </RatingItem>
                      <RatingItem>
                        <Rating>{rank}</Rating>
                        <RatingTitle>Rank</RatingTitle>
                      </RatingItem>
                      <RatingItem>
                        <Rating>{popularity}</Rating>
                        <RatingTitle>Popularity</RatingTitle>
                      </RatingItem>
                    </RatingBlock>
                    
could be simplified:
const ratings = [
  { value: score, title: 'Score' },
  { value: rank, title: 'Rank' },
  { value: popularity, title: 'Popularity' },
]

then:
{ratings.map(({ value, title }) => (
                      <RatingItem>
                        <Rating>{value}</Rating>
                        <RatingTitle>{title}</RatingTitle>
                      </RatingItem>
))}




so in this case if you would need to add one more html tag here (lets say average rating) you would only need to change it in 1 place instead of 3, only here:
{ratings.map(({ value, title**, averageRating** }) => (
                      <RatingItem>
                        <Rating>{value}</Rating>
                        <RatingTitle>{title}</RatingTitle>
                        **<AverageRating>{averageRating}</AverageRating>**
                      </RatingItem>
))}
but with the current structure, you would need to do it in 3 places:

                    <RatingBlock>
                      <RatingItem>
                        <Rating>{score}</Rating>
                        <RatingTitle>Score</RatingTitle>
                        **<AverageRating>{averageRating}</AverageRating>**
                      </RatingItem>
                      <RatingItem>
                        <Rating>{rank}</Rating>
                        <RatingTitle>Rank</RatingTitle>
                        **<AverageRating>{averageRating}</AverageRating>**
                      </RatingItem>
                      <RatingItem>
                        <Rating>{popularity}</Rating>
                        <RatingTitle>Popularity</RatingTitle>
                        **<AverageRating>{averageRating}</AverageRating>**
                      </RatingItem>
                    </RatingBlock>

and in case you would have more than 3 items - it would cause even more problems. So try to use .map in case you see some repeating items

Reply:
pls check if I did it right

Reply:
yep. Moving it to separate component is also good decision